### PR TITLE
Convert GetAllMountPoints to UnmanagedCallersOnly

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

File: src/libraries/Common/src/Interop/Windows/HttpApi/Interop.HttpApi.cs [hunk 1]
```diff
@@ -459,7 +459,7 @@ internal struct HTTP_BINDING_INFO
         internal static partial uint HttpInitialize(HTTPAPI_VERSION version, uint flags, IntPtr pReserved);
 
         [LibraryImport(Libraries.HttpApi, SetLastError = true)]
-        internal static partial uint HttpSetUrlGroupProperty(ulong urlGroupId, HTTP_SERVER_PROPERTY serverProperty, IntPtr pPropertyInfo, uint propertyInfoLength);
+        internal static unsafe partial uint HttpSetUrlGroupProperty(ulong urlGroupId, HTTP_SERVER_PROPERTY serverProperty, void* pPropertyInfo, uint propertyInfoLength);
```
Comment (jkotas) [inline src/libraries/Common/src/Interop/Windows/HttpApi/Interop.HttpApi.cs:462]: Unrelated change that I had around

File: src/libraries/Common/src/Interop/Unix/System.Native/Interop.MountPoints.cs [hunk 2]
```diff
@@ -2,37 +2,54 @@
...
-        private unsafe delegate void MountPointFound(byte* name);
+        [LibraryImport(Libraries.SystemNative, EntryPoint = "SystemNative_GetAllMountPoints")]
+        private static unsafe partial int GetAllMountPoints(delegate* unmanaged<void*, byte*, void> onFound, void* context);
 
-        [LibraryImport(Libraries.SystemNative, EntryPoint = "SystemNative_GetAllMountPoints", SetLastError = true)]
-        private static partial int GetAllMountPoints(MountPointFound mpf);
+        private struct AllMountPointsContext
+        {
+            public List<string> _results;
+            public ExceptionDispatchInfo? _exception;
```
Comment (am11) [inline src/libraries/Common/src/Interop/Unix/System.Native/Interop.MountPoints.cs:20]: nit: PacalCase for public fields

File: src/libraries/Common/src/Interop/Windows/HttpApi/Interop.HttpApi.cs (hunk 1 above)
Comment (am11) [reply src/libraries/Common/src/Interop/Windows/HttpApi/Interop.HttpApi.cs:462]: Just curious, does it save an extra validation? I always thought `SafeHandle` is expensive for P/Invokes and shouldn't be used unless native method requires it, but `IntPtr` in my mind was as good as `void*`. :)

Comment (teo-tsirpanis) [reply src/libraries/Common/src/Interop/Windows/HttpApi/Interop.HttpApi.cs:462]: It's mostly for style reasons; now that `IntPtr` === `nint`, let's use an actual pointer type instead of a native integer when we pass a pointer. I don't think there are differences in execution behavior.

File: src/libraries/Common/src/Interop/Unix/System.Native/Interop.MountPoints.cs [hunk 3]
```diff
@@ -2,37 +2,54 @@
...
-                    }
-                    found[count++] = Marshal.PtrToStringAnsi((IntPtr)name)!;
-                });
+                GetAllMountPoints(&AddMountPoint, Unsafe.AsPointer(ref context));
             }
 
-            Array.Resize(ref found, count);
-            return found;
+            if (context._exception != null)
+                context._exception.Throw();
```
Comment (stephentoub) [inline src/libraries/Common/src/Interop/Unix/System.Native/Interop.MountPoints.cs:50]:
```suggestion
            context._exception?.Throw();
```

**Review by stephentoub — APPROVED** [review]

File: src/libraries/Common/src/Interop/Unix/System.Native/Interop.MountPoints.cs [hunk 4]
```diff
@@ -2,37 +2,54 @@
...
+        {
+            public List<string> _results;
+            public ExceptionDispatchInfo? _exception;
+        }
+
+        [UnmanagedCallersOnly]
+        private static unsafe void AddMountPoint(void* context, byte* name)
+        {
+            ref AllMountPointsContext callbackContext = ref Unsafe.As<byte, AllMountPointsContext>(ref *(byte*)context);
+            callbackContext._results = new List<string>();
```
Comment (stephentoub) [inline src/libraries/Common/src/Interop/Unix/System.Native/Interop.MountPoints.cs:27]:
```suggestion
```

File: src/libraries/Common/src/Interop/Unix/System.Native/Interop.MountPoints.cs (hunk 2 above)
Comment (jkotas) [inline src/libraries/Common/src/Interop/Unix/System.Native/Interop.MountPoints.cs:20]:
```suggestion
            internal List<string> _results;
            internal ExceptionDispatchInfo? _exception;
```
